book/src/Setup.md

```diff
@@ -1,65 +1,88 @@
-  is enabled on kubelet (kubelet flag KubeletPluginsWatcher is set), then this option
-  should be set.
-```
+The CSI driver discovery uses the [Kubelet Plugin Watcher][plugin-watcher] feature which allows Kubelet to discover deployed CSI drivers automatically.  The registar sidecar pod exposes an internal registration server via a Unix data socket path. The Kubelet monitors for the path and once detected, the Kubelet contacts the registrar to query driver information.
 
-In addition, in order to expose the registration socket to the host, the
-`driver-registrar` side-car container must have a HostPath volume, with the
-kubelet plugin directory as the host path (typically at
-`/var/lib/kubelet/plugins/`), mounted at `/registration` in the container.
+Using this feature will not have any effect on how drivers behave, however, this will be the way CSI works internally in coming releases.  
 
-```yaml
-volumeMounts:
-- name: registration-dir
-  mountPath: /registration
 
+#### Registrar sidecar configuration
+To enable the Kubelet to query information about your CSI driver, you will need to configure your registrar sidecar container properly as show below:
+
+```yaml
+- name: driver-registrar
+    args:
+    - --v=5
+    - --csi-address=/csi/csi.sock
+    - --kubelet-registration-path=/var/lib/kubelet/plugins/<driver-name>/csi.sock
+    env:
+    - name: KUBE_NODE_NAME
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: spec.nodeName
+    image: quay.io/k8scsi/driver-registrar:v0.2.0
+    imagePullPolicy: Always
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    volumeMounts:
+    - name: registration-dir
+      mountPath: /registration
 ...
 volumes:
-- name: registration-dir
-  hostPath:
+  name: socket-dir
+  - hostPath:
+      path: /var/lib/kubelet/plugins/csi-hostpath
+      type: DirectoryOrCreate
+  name: registration-dir
+  - hostPath:
       path: /var/lib/kubelet/plugins
       type: Directory
 ```
 
+Where:
+
+- `--csi-address` - specifies the Unix domain socket path, on the host, for the CSI driver.  This allows the registrar sidecar to communicate with the driver to get discovery information.
+- `--kubelet-registration-path` - enables automatic driver discovery.  It specifyies a Unix domain socket path, on the host, for the registrar sidecar container's regisgtration service handler.  This path must be a location where the Kubelet Plugin Registration watcher monitors for new CSI plugins. Once discovered, the Kubelet plugin mechanism will communicate with the registrar sidecar to retrieve CSI driver information.
+
+
+In addition, in order to expose the registration socket to the host, 
+the `driver-registrar` side-car container must have a `HostPath` volume, with the
+kubelet plugin directory as the host path (typically at
+`/var/lib/kubelet/plugins/`), mounted at `/registration` in the container as shown above.
```

Comment:
Is there a way to avoid hardcoding the `/var/lib/kubelet/plugins/` in the .yaml file?

Reply:
@pohly that value should coincide with whatever the Kubelet plugin directory is set as.  I would not call it a hard-coded value. Unfortunately, I don't think there's a way to read that programmatically or via environment variable.